src/components/Layout/PhoneNumberInput.tsx

```diff
@@ -63,6 +66,32 @@ export const PhoneNumberInput: FunctionComponent<{
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   onSubmit = () => {}
 }) => {
+  const formatPhoneNumber = (
+    mobileNumberValue: string,
+    region = "SG"
+  ): string => {
+    try {
+      return phoneUtil.format(
+        phoneUtil.parse(mobileNumberValue, region),
+        PhoneNumberFormat.NATIONAL
+      );
+    } catch (e) {
+      if (mobileNumber && mobileNumber?.length < mobileNumberValue.length) {
+        return mobileNumber;
+      }
+      return mobileNumberValue;
+    }
+  };
+  const [mobileNumber, setMobileNumber] = useState<string>(
+    formatPhoneNumber(mobileNumberValue.replace(" ", ""))
+  );
+
+  const onHandleChangePhoneNumber = (text: string): void => {
+    const newMobileValue = formatPhoneNumber(text);
+    setMobileNumber(newMobileValue);
+    const cleanText: string = newMobileValue.replace(" ", "");
+    onChangeMobileNumber(cleanText);
+  };
```

Comment:
Just some cleaning up:

```suggestion
  const onHandleChangePhoneNumber = (text: string): void => {
    const cleanText: string = formatPhoneNumber(text).replace(" ", "");
    onChangeMobileNumber(cleanText);
  };
```